src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,32 @@
+package command;
+import oracle.TaskList;
+import oracle.Ui;
+import entry.Task;
+
+public class DeleteCommand implements Command {
+    private final int taskIndex;
+
+    /**
+     * @param i index of the task to be deleted
+     */
+    public DeleteCommand(int i) {
+        this.taskIndex = i;
+    }
+
+    /**
+     * @param ui helper to print the new deleted task
+     * @param tasks we call delete on the indicated task in this TaskList
+     * @return true
+     */
+    @Override
+    public boolean execute(Ui ui, TaskList tasks) {
+        try {
+            Task t = tasks.get(this.taskIndex);
+            tasks.remove(this.taskIndex);
```

Comment:
Task t = tasks.remove(this.taskIndex)

I think can make this simpler by combining both statements.

Reply:
good idea, thanks